server/src/weaverbird/backends/pypika_translator/translators/base.py

```diff
@@ -238,45 +250,129 @@ def aggregate(
         step: "AggregateStep",
     ) -> StepContext:
         agg_selected: list[Field] = []
+        window_selected: list[tuple[int, Field]] = []
+        window_subquery_list: list[Tables] = []
+
+        def _build_window_subquery() -> Any:
+            min_window_index = min([c[0] for c in window_selected])
+            first_wq = Table(f'wq{min_window_index}')
+            merged_query = (
+                self.QUERY_CLS.from_(window_subquery_list[0])
+                .select(
+                    *step.on,
+                    *[
+                        getattr(first_wq, col[1].alias)
+                        for col in window_selected
+                        if col[0] == min_window_index
+                    ],
+                )
+                .as_('window_subquery')
+            )
+            if len(window_subquery_list) > 1:
+                for index, sq in enumerate(window_subquery_list[1:]):
+                    wq_temp = Table(f'wq{min_window_index + index + 1}')
+                    merged_query = (
+                        merged_query.join(sq)
+                        .on_field(*step.on)
+                        .select(
+                            *[
+                                getattr(wq_temp, col[1].alias)
+                                for col in window_selected
+                                if col[0] == min_window_index + index + 1
+                            ]
+                        )
+                        .as_('window_subquery')
```

Comment:
Will having the same alias several times work if there are several window subqueries ?

Maybe a fixture with multiple window functions could be interesting here ? Something like
```json
{
            "newcolumns": [
              "first_name",
              "first_beer_kind"
            ],
            "aggfunction": "first",
            "columns": [
              "name",
              "beer_kind"
            ]
          }
```

Reply:
There's already a fixture for that `first_two_cols_pypika`